tests/test_personal_names.py

```diff
@@ -44,56 +44,65 @@ def test_display_name_to_sort_name(self):
+            # because it's in "Family, Given" format or for some other
+            # reason.
+            eq_(x, m(x))
+        unchanged(u"Bitshifter, Bob")
+        unchanged(u"Prince")
+        unchanged(u"Pope Francis")
+        unchanged(u"Lord Byron")
+        unchanged(u"Heliodorus (bp. of Tricca.)")
+        unchanged(u"谢新源 (Xie Xinyuan)")
+
+        sort_name = m(u"Bob Bitshifter")
         eq_(u"Bitshifter, Bob", sort_name)
 
         # foreign characters don't confuse the algorithm
-        sort_name = display_name_to_sort_name(u"Боб Битшифтер")
+        sort_name = m(u"Боб Битшифтер")
         eq_(u"Битшифтер, Боб", sort_name)
 
-        sort_name = display_name_to_sort_name(u"Bob Bitshifter, Jr.")
+        sort_name = m(u"Bob Bitshifter, Jr.")
         eq_(u"Bitshifter, Bob Jr.", sort_name)
 
-        sort_name = display_name_to_sort_name(u"Bob Bitshifter, III")
+        sort_name = m(u"Bob Bitshifter, III")
         eq_(u"Bitshifter, Bob III", sort_name)
 
-        # already having a comma still gets good results
-        sort_name = display_name_to_sort_name(u"Bob, The Grand Duke of Awesomeness")
-        eq_(u"Bob, Duke of Awesomeness The Grand", sort_name)
+        eq_("Beck, James M. (James Montgomery)",
+            m("James M. (James Montgomery) Beck"))
 
         # all forms of PhD are recognized
-        sort_name = display_name_to_sort_name(u"John Doe, PhD")
+        sort_name = m(u"John Doe, PhD")
         eq_(u"Doe, John PhD", sort_name)
-        sort_name = display_name_to_sort_name(u"John Doe, Ph.D.")
+        sort_name = m(u"John Doe, Ph.D.")
         eq_(u"Doe, John PhD", sort_name)
-        sort_name = display_name_to_sort_name(u"John Doe, Ph D")
+        sort_name = m(u"John Doe, Ph D")
         eq_(u"Doe, John PhD", sort_name)
-        sort_name = display_name_to_sort_name(u"John Doe, Ph. D.")
+        sort_name = m(u"John Doe, Ph. D.")
         eq_(u"Doe, John PhD", sort_name)
-        sort_name = display_name_to_sort_name(u"John Doe, PHD")
+        sort_name = m(u"John Doe, PHD")
         eq_(u"Doe, John PhD", sort_name)
 
-        sort_name = display_name_to_sort_name(u"John Doe, M.D.")
+        sort_name = m(u"John Doe, M.D.")
         eq_(u"Doe, John MD", sort_name)
 
-        # corporate name is unchanged
-        sort_name = display_name_to_sort_name(u"Church of Jesus Christ of Latter-day Saints")
-        eq_(u"Church of Jesus Christ of Latter-day Saints", sort_name)
+        # corporate names are unchanged
+        unchanged(u"Church of Jesus Christ of Latter-day Saints")
+        unchanged(u"(C) 2006 Vanguard")
 
+        # NOTE: These results are not the best.
+        eq_("Pope XVI, Benedict", m(u"Pope Benedict XVI"))
+        eq_("Lord Alfred, Tennyson", m(u"Alfred, Lord Tennyson"))
+        eq_("Lord Lennox, William", m("Lord William Lennox"))
+        sort_name = m(u"Bob, The Grand Duke of Awesomeness")
+        eq_(u"The Grand Bob, Duke of Awesomeness", sort_name)
```

Comment:
In these, we still get a wrong answer, but the wrong answer is slightly different. 
The only one here that's not new is "Bob, The Grand Duke of Awesomeness", which was previously normalized to "Bob, Duke of Awesomeness The Grand".

The difference is that "The Grand" is now treated as a title, and titles like "Pope" and "Lord" and "The Grand" now go at the front of someone's sort name.